datafusion-cli/src/main.rs

```diff
@@ -87,6 +96,13 @@ struct Args {
         help = "Reduce printing other than the results and work quietly"
     )]
     quiet: bool,
+
+    #[clap(
+        long,
+        help = "Specify the memory pool type 'greedy' or 'fair', default to 'greedy'",
+        validator(is_valid_memory_pool_type)
+    )]
+    mem_pool_type: Option<String>,
```

Comment:
Another, perhaps more ideomatic way to do this, is to define an enum like

```suggestion
    mem_pool_type: Option<PoolType>,
```


```rust
enum PoolType {
  Greedy,
  Fair
}

impl FromStr for PoolType { 
...
}
```

And then I think clap will do the right thing